stat/distuv/triangle.go

```diff
@@ -0,0 +1,177 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// UnitTriangle is an instantiation of the triangle distribution with A=0, C=0.5, and B=1
+var UnitTriangle = Triangle{A: 0, C: 0.5, B: 1}
+
+// Triangle represents a triangle distribution (https://en.wikipedia.org/wiki/Triangular_distribution).
+type Triangle struct {
+	A      float64
+	C      float64
+	B      float64
+	Source *rand.Rand
+}
+
+// CDF computes the value of the cumulative density function at x.
+func (t Triangle) CDF(x float64) float64 {
+	if x <= t.A {
```

Comment:
This would be clearer as a switch, also, for n^2, please just multiply.

```
switch {
case x <= t.A:
    return 0
case x <= t.C:
    d := x-t.A
    return (d*d) / ((t.B - t.A) * (t.C - t.A))
case x < t.B:
    d := x-t.B
    return 1 - (d*d)/((t.B-t.A)*(t.B-t.C))
default:
    return 1
}
```